### DISCUSSION TITLE
Take over form option picker values of table A in table B without relation fields

### DISCUSSION THREAD

**Discussion by JohannesM007** [Budibase/budibase#9694] [Help] [answered]
Hello everyone,
I am pretty new here and it try to creat a complaint app ;-).
There is one table named “Complaints“ and a second table named “Reasons“.
There is no relationship between the two tables yet.
Screen route =  /complaints/new/row
In the complaint form, I would like to transfer the selected reasons to the complaint table in the end.

The reason form is embedded in the complaint form.
I have created three "Option Pickers" to use the reasons records in dependence: Channel – Reason - Subreason
When the “Channel“ is selected, the next option field „Reason“  is bind via data provider to {{ Form Reason.Fields.Channel }}
The option field “Subreason” is bind to a second data provider {{ Form Reason.Fields.Reason }}
This works so far.

Unfortunately, the fields not taken over into the complaint table. 
The columns in the complaint table are text fields with the same column name like in the “Reason” table.

However, in the Reason form (Type: Create) the selected field values will be gone when I go to the next form step, and will not been saved. 
How can I take over the values to the complaint table fields? I can't find a similar case here.
Thank you in advance.
Johannes 

![grafik](https://user-images.githubusercontent.com/125383022/218785831-39eebe6f-bb7d-4aa3-9949-fde3e44e7c52.png)

![grafik](https://user-images.githubusercontent.com/125383022/218786101-21104cda-5a4a-469f-aa65-7d8921543e03.png)

![grafik](https://user-images.githubusercontent.com/125383022/218787749-71d1fd7c-1072-4531-b06d-83b6059a4ff3.png)

![grafik](https://user-images.githubusercontent.com/125383022/218788144-543c8736-9623-4cb7-a235-281557518990.png)

**Comment by melohagan** [comment]
Hi @JohannesM007 

Are you using the **Save Row** action on your 'Next' button?

**Comment by JohannesM007** [comment]
Hi @melohagan 
No there is no Save Row action in this next button on form step2!

![grafik](https://user-images.githubusercontent.com/125383022/219405456-d1d9c6e1-b31e-4491-bccc-e1567d834e9c.png)

There is only a save button on Step 3 and I was playing with some "Save Row" settings, but if i go back and forth, the option picers will be empty again.
Is there an easy way to save the values from the option picers in step 2 and then pass them to the complaint 
table via the save button in step 3.

![grafik](https://user-images.githubusercontent.com/125383022/219394753-c792a751-177f-4200-86d2-6c993dd40be7.png)

![grafik](https://user-images.githubusercontent.com/125383022/219394124-5aded39b-715c-43b5-ac8c-7fd313023d8d.png)

**Reply by melohagan** [reply]
Hey @JohannesM007 Would you be able to share your app export to support@budibase.com ? It would help to get a look at the app!

**Reply by JohannesM007** [reply]
Hi @melohagan
the app has been send to the support email right now!
Tx
Johannes

**Reply by melohagan** [reply]
Hey @JohannesM007 The attachment didn't come through in the email, it may have been stripped. 

> Executable attachments were removed from this message. Contact your IT Admin if the files were expected and needed.

Could you possibly email me a file download link instead?

**Comment by JohannesM007** [comment]
Sorry,
the link has been send to support@budibase.com 
Cheers Johannes

**Comment by melohagan** [comment, marked as answer]
Hey @JohannesM007 

Thank you for the app export, I think I have a solution for you.

For each of the dropdowns (Channel, Reason and Subreason) add an **On Change** action, and provide the `{{ Field Value }}` binding.

For example:
<img width="1133" alt="Screenshot 2023-02-17 at 14 13 20" src="https://user-images.githubusercontent.com/101575380/219678705-eae1a4dd-4d63-4a08-910a-8780b0f35a37.png">

Next set the default value for each of the dropdowns to the state value, e.g.
<img width="236" alt="Screenshot 2023-02-17 at 14 15 16" src="https://user-images.githubusercontent.com/101575380/219679101-013106eb-9d17-401c-8f74-d9864a772bea.png">

**Result**
![new_complaint](https://user-images.githubusercontent.com/101575380/219679380-0b1dc205-3ec2-41c5-8860-f6127844ab81.gif)

Change the **Save Row** action to use the app state as well:
![Screenshot 2023-02-17 at 14 21 35](https://user-images.githubusercontent.com/101575380/219680468-f896af5f-7b8d-40cc-a867-6c7aed6d50f7.png)

Data output:
![Screenshot 2023-02-17 at 14 22 52](https://user-images.githubusercontent.com/101575380/219680748-2d51e64b-3d6f-4235-9c05-d94ea5854521.png)

![Screenshot 2023-02-17 at 14 23 08](https://user-images.githubusercontent.com/101575380/219680809-450f0b87-485a-49da-bd6c-af1d019a0698.png)

**Comment by JohannesM007** [comment]
Hey @melohagan,
wow really cool - thank you so much for the quick reply!
It's working

I have added just some more steps in the savebutton to delete the state values for the next run ;-)
![grafik](https://user-images.githubusercontent.com/125383022/219699727-07bdab7f-1a4d-436e-aee6-942e3a044781.png)

All the best
Johannes
